packages/dds/tree/src/shared-tree/treeAlpha.ts

```diff
@@ -481,6 +513,72 @@ export const TreeAlpha: TreeAlpha = {
 		const parentSchema = treeNodeApi.schema(parent);
 		return getPropertyKeyFromStoredKey(parentSchema, storedKey);
 	},
+
+	child: (node: TreeNode, key: string | number): TreeNode | TreeLeafValue | undefined => {
+		const flexNode = getOrCreateInnerNode(node);
+		debugAssert(() => !flexNode.context.isDisposed() || "FlexTreeNode is disposed");
+
+		const schema = treeNodeApi.schema(node);
+		const storedKey = tryGetStoredKeyFromPropertyKey(schema, key);
```

Comment:
given you have to special case arrays after this anyway, making tryGetStoredKeyFromPropertyKey support arrays probably doesn't make sense.

Also having tryGetStoredKeyFromPropertyKey support arrays is a bit odd since you really wand a FieldKey back from such a util for all non array cases, but on array nodes the field key is always emptyKey. I don't think providing a unified helper here makes sense because of that

Reply:
Ah, yeah that makes sense. This was added before I added array handling. Will inline as appropriate.